source/ldap.d

```diff
@@ -456,20 +456,25 @@ struct LDAPConnection
 			{
 				string attr_name = attr.to!string;
 
-				auto ppValue = ldap_get_values(_handle, entry, attr);
+				auto ppValue = ldap_get_values_len(_handle, entry, attr);
 				if (!ppValue)
 				{
 					result.attributes[attr_name] = [];
 				}
 				else
 				{
-					auto iValue = ldap_count_values(ppValue);
+					auto iValue = ldap_count_values_len(ppValue);
 					if (!iValue)
 						result.attributes[attr_name] = [];
 					else
-						result.attributes[attr_name] = ppValue[0 .. iValue].map!(a => a.to!(char[])
-								.idup).array;
-					ldap_value_free(ppValue);
+                        result.attributes[attr_name] = 
+                            ppValue[0 .. iValue].map!(a =>
+                                attr_name == "thumbnailPhoto" ?
```

Comment:
why only for thumbnailPhoto and not in general for everything?

Reply:
For some reason I thought that for binary data to do so. Hmm.. I'll fix it now.